XCode15 / iOS17: Add 3DS Support through EmuThreeds / Update Play! PS2 to latest code

rf2222222 opened the pull request:
### What does this PR do
   * Added emuthreed to play 3DS games 
   * Added 3DS touch screen controller
   * Update Play! to latest GitHub (provides improved sound compared to before)
   * Prevent game from being moved into conflicts folder during artwork download
   * Added RxRealm as library with XCode15 Beta6 patches
   * Update Flycast to latest GitHub tree, with Vulkan support
   * Added Sound Volume Control Option to RetroArch, Emuthreeds, Play, etc
   * LZH / RAR / ZIP archive load support for NP2Kai to play PC98 games with boot disk from archive (extracted files saved to Battery States)
   * Zip archive load / Play support for DosBox / NeoGeo / MAME
   * Optimized ROM import w/ Caching in RomDatabase, added 'Refresh' button to scan / update the game library w/ new rom
   * Various updates to GameLibrary interfaces to work / appear ok on iOS17
   * Added Apple IIe core support via MAME
   * Added MAME / NeoGeo touchscreen controller
   * Vulkan is now the default graphics renderer on PPSSPP
   

On iOS17 AltKit does not work, so to enable JIT:
  1. Launch Provenance XL Release on iPhone/iPad, Launch XCode15, select the iPhone/iPad
  2. Debug -> Attach to Process by PID or Name
  3. Enter "Provenance XL" (assuming Provenance XL Release is installed), Click Attach. After confirming the game (PS2, Wii, etc) is working, Debug -> Detach from Provenance XL

### Where should the reviewer start
emuthreed (should play) -> Play!

### How should this be manually tested
Play .3DS games with emuthreed, Play! offers more games compatibility

### Any background context you want to provide

### What are the relevant tickets
https://github.com/Provenance-Emu/Provenance/issues/2201

### Screenshots (important for UI changes)
![IMG_0930](https://github.com/Provenance-Emu/Provenance/assets/6903427/5808b40f-c1da-4ccb-9434-8ce86abbf2ad)
![IMG_0934](https://github.com/Provenance-Emu/Provenance/assets/6903427/18b28f08-0827-4853-b0bf-bda4681780af)
![IMG_0936](https://github.com/Provenance-Emu/Provenance/assets/6903427/9c8e0dd2-a9ca-4af8-8c36-4e548e6f0b9c)

### Questions

BluStik commented:
Hey, when I turn my device into landscape mode using the EmuThreeds core, it automatically turns my phone back to portrait mode.

BluStik commented:
Hey, I found one more bug. It is with the joystick. 

https://github.com/Provenance-Emu/Provenance/assets/120103991/cdf7e6af-4cb5-400c-beb9-da772acb4ea1

rf2222222 commented:
> Hey, I found one more bug. It is with the joystick in EmuThreeds. When I pull it past a certain point it stops working.
> 
>  0823.1.mov

This is fixed in the latest push =]

BluStik commented:
(Should be the last thing) 
Might just be a SideStore issue with Xcode 15, but it gives me this error when I try to install. It works great with sideloady though.
![IMG_0967](https://github.com/Provenance-Emu/Provenance/assets/120103991/633d7664-1d7a-4218-b4a4-9a9efe4886e4)

rf2222222 commented:
> (Should be the last thing) Might just be a SideStore issue with Xcode 15, but it gives me this error when I try to install. It works great with sideloady though. ![IMG_0967](https://user-images.githubusercontent.com/120103991/263123755-633d7664-1d7a-4218-b4a4-9a9efe4886e4.jpeg)

This should be fixed in the latest push =]

Spidy123222 commented:
There is a PR in moltenvk for geometry shaders and I wonder if it’s possible to use it for emuthreeds. The dev told me it should help performance but I wonder how that would be here. The emulator already only supports iOS 16+ so it makes sense.

rf2222222 commented:
> There is a PR in moltenvk for geometry shaders and I wonder if it’s possible to use it for emuthreeds. The dev told me it should help performance but I wonder how that would be here. The emulator already only supports iOS 16+ so it makes sense. [KhronosGroup/MoltenVK#1815](https://github.com/KhronosGroup/MoltenVK/pull/1815)
> 
> also some now psp buttons is now odd on touch ui.

This should be fixed in the latest push =] (the ppsspp buttons / retroarch ppsspp core loading menu)

Adding pipeline threading to geometry shader sounds good. There’s an implementation by the author around pipelining in general, but is minimally used at the moment due to stability / hardware GPU compatibility with older iOS devices. If MoltenVK provides good features around this we could definitely add to this when it becomes release code =] (Which.. Citra / emuThreeds teams seem to be very much on top of, so we should get an update from the teams)

rf2222222 commented:
> > There is a PR in moltenvk for geometry shaders and I wonder if it’s possible to use it for emuthreeds. The dev told me it should help performance but I wonder how that would be here. The emulator already only supports iOS 16+ so it makes sense. [KhronosGroup/MoltenVK#1815](https://github.com/KhronosGroup/MoltenVK/pull/1815)
> > also some now psp buttons is now odd on touch ui.
> 
> This should be fixed in the latest push =] (the ppsspp buttons / retroarch ppsspp core loading menu)
> 
> Adding pipeline threading to geometry shader sounds good. There’s an implementation by the author around pipelining in general, but is minimally used at the moment due to stability / hardware GPU compatibility with older iOS devices. If MoltenVK provides good features around this we could definitely add to this when it becomes release code =] (Which.. Citra / emuThreeds teams seem to be very much on top of, so we should get an update from the teams)

The author of emuThreeds deserves lots of coffee from everyone, here: https://github.com/emuPlace/emuThreeDS It’s truly a great port =]

Dix01 commented:
first of all thank you for your hard work.
I noticed a bug with the Emuthreeds core.
It runs perfectly whenever i load a .3ds file for a while, however after some time when i relaunch it whether with or without JIT, it simply crashes. All other emulator cores work flawlessly. Emuthreeds crashes when relaunched after some time, and the only way for it to work again, is to completely delete Provenance and re-sideload it.

rf2222222 commented:
> first of all thank you for your hard work. I noticed a bug with the Emuthreeds core. It runs perfectly whenever i load a .3ds file for a while, however after some time when i relaunch it whether with or without JIT, it simply crashes. All other emulator cores work flawlessly. Emuthreeds crashes when relaunched after some time, and the only way for it to work again, is to completely delete Provenance and re-sideload it. @rf2222222

This sounds like a memory leak issue. The game reserves a good chunk of memory for Fastmem access and reserved space for save states. The memory should free up when closing the App though, so closing and reopening Provenance should free up the memory, all the games should then work fine thereafter. 

To look into this further, I need:
   * Device type (e.g. 2020 iPad w/ iOS17)
   * Does this happen after closing / opening Provenance?
   * Does this happen after particular action (e.g. loading / saving game states x number of times, resetting / restarting / reloading the game x number of times ..etc)?

Games becoming unstable after a number of restarts is a common problem depending on the memory availability (especially with JIT), so the best remedy for these is to close / open Provenance and it usually frees up all the memory to play the game fresh again.

Dix01 commented:
> > first of all thank you for your hard work. I noticed a bug with the Emuthreeds core. It runs perfectly whenever i load a .3ds file for a while, however after some time when i relaunch it whether with or without JIT, it simply crashes. All other emulator cores work flawlessly. Emuthreeds crashes when relaunched after some time, and the only way for it to work again, is to completely delete Provenance and re-sideload it. @rf2222222
> 
> 
> 
> This sounds like a memory leak issue. The game reserves a good chunk of memory for Fastmem access and reserved space for save states. The memory should free up when closing the App though, so closing and reopening Provenance should free up the memory, all the games should then work fine thereafter. 
> 
> 
> 
> To look into this further, I need:
> 
>    * Device type (e.g. 2020 iPad w/ iOS17)
> 
>    * Does this happen after closing / opening Provenance?
> 
>    * Does this happen after particular action (e.g. loading / saving game states x number of times, resetting / restarting / reloading the game x number of times ..etc)?
> 
> 
> 
> Games becoming unstable after a number of restarts is a common problem depending on the memory availability (especially with JIT), so the best remedy for these is to close / open Provenance and it usually frees up all the memory to play the game fresh again. 


Thank you for the fast reply. 

Im using iPhone 13 Pro iOS 16.5.1

I had assumed it was a memory leak as well, however closing and reopening the app does not fix the issue, it forever crashes from then on whenever i try loading a 3ds file.

What i did was play Pokemon a little bit, then just switched to a different app for a little while. That's it, basically any "long absence" from the game makes it crash and not relaunch ever again, unless i delete the app and reinstall it

Dix01 commented:
Ok i did some more testing and apparently it works okay when i do not load state saves. Have to reset the ROM and start normally for it to not crash

rf2222222 commented:
> Ok i did some more testing and apparently it works okay when i do not load state saves. Have to reset the ROM and start normally for it to not crash

Sometimes when the app goes to background, but game crashed in the background, the auto save that was created around that time creates bad save data. With this, the game needs to be run from ‘Restart’ rather than ‘Continue’ button that tries to auto load the latest save data. Creating new save data will solve the issue thereafter.

I added a quick fix that ensures the core is paused before being sent to background, and core’s shutdown is called prior to app termination (which frees up the memory). With this, the issue should be fixed =]

rf2222222 commented:
> > Ok i did some more testing and apparently it works okay when i do not load state saves. Have to reset the ROM and start normally for it to not crash
> 
> Sometimes when the app goes to background, but game crashed in the background, the auto save that was created around that time creates bad save data. With this, the game needs to be run from ‘Restart’ rather than ‘Continue’ button that tries to auto load the latest save data. Creating new save data will solve the issue thereafter.
> 
> I added a quick fix that ensures the core is paused before being sent to background, and core’s shutdown is called prior to app termination (which frees up the memory). With this, the issue should be fixed =]

I added a feature that auto saves the game state when the app goes to background, with the game’s ‘Continue’ button during launch restarting from the latest auto saved state. With this, the issue should be fixed in the latest push =]

Dix01 commented:
> > > Ok i did some more testing and apparently it works okay when i do not load state saves. Have to reset the ROM and start normally for it to not crash
> 
> > 
> 
> > Sometimes when the app goes to background, but game crashed in the background, the auto save that was created around that time creates bad save data. With this, the game needs to be run from ‘Restart’ rather than ‘Continue’ button that tries to auto load the latest save data. Creating new save data will solve the issue thereafter.
> 
> > 
> 
> > I added a quick fix that ensures the core is paused before being sent to background, and core’s shutdown is called prior to app termination (which frees up the memory). With this, the issue should be fixed =]
> 
> 
> 
> I added a feature that auto saves the game state when the app goes to background, with the game’s ‘Continue’ button during launch restarting from the latest auto saved state. With this, the issue should be fixed in the latest push =]

Also found another bug! Apparently controller opacity does not change at all. Always resets to default!

honorsea commented:
I 

> Builds for this Pull Request are available at
> [Provenance.ipa](https://github.com/Provenance-Emu/Provenance/suites/15531578798/artifacts/886567416)
> [ProvenanceTV.ipa](https://github.com/Provenance-Emu/Provenance/suites/15531578798/artifacts/886567418)
> [Provenance-UnderDevelopment.ipa](https://github.com/Provenance-Emu/Provenance/suites/15531578798/artifacts/886567419)
> [Provenance-XL.ipa](https://github.com/Provenance-Emu/Provenance/suites/15531578798/artifacts/886567420)
> Have a nice day.

I believe this update broke the touch-screen functionality. It’s now working once or twice out of ten times where before it was working all the time

rf2222222 commented:
> > > > Ok i did some more testing and apparently it works okay when i do not load state saves. Have to reset the ROM and start normally for it to not crash
> > 
> > 
> > > 
> > 
> > 
> > > Sometimes when the app goes to background, but game crashed in the background, the auto save that was created around that time creates bad save data. With this, the game needs to be run from ‘Restart’ rather than ‘Continue’ button that tries to auto load the latest save data. Creating new save data will solve the issue thereafter.
> > 
> > 
> > > 
> > 
> > 
> > > I added a quick fix that ensures the core is paused before being sent to background, and core’s shutdown is called prior to app termination (which frees up the memory). With this, the issue should be fixed =]
> > 
> > 
> > I added a feature that auto saves the game state when the app goes to background, with the game’s ‘Continue’ button during launch restarting from the latest auto saved state. With this, the issue should be fixed in the latest push =]
> 
> Also found another bug! Apparently controller opacity does not change at all. Always resets to default!

This should now be fixed in the latest push =]

rf2222222 commented:
> > Builds for this Pull Request are available at
> > [Provenance.ipa](https://github.com/Provenance-Emu/Provenance/suites/15531578798/artifacts/886567416)
> > [ProvenanceTV.ipa](https://github.com/Provenance-Emu/Provenance/suites/15531578798/artifacts/886567418)
> > [Provenance-UnderDevelopment.ipa](https://github.com/Provenance-Emu/Provenance/suites/15531578798/artifacts/886567419)
> > [Provenance-XL.ipa](https://github.com/Provenance-Emu/Provenance/suites/15531578798/artifacts/886567420)
> > Have a nice day.
> 
> I believe this update broke the touch-screen functionality. It’s now working once or twice out of ten times where before it was working all the time

This should now be fixed in the latest push =]

Spidy123222 commented:
> > There is a PR in moltenvk for geometry shaders and I wonder if it’s possible to use it for emuthreeds. The dev told me it should help performance but I wonder how that would be here. The emulator already only supports iOS 16+ so it makes sense. [KhronosGroup/MoltenVK#1815](https://github.com/KhronosGroup/MoltenVK/pull/1815)
> 
> > 
> 
> > also some now psp buttons is now odd on touch ui.
> 
> 
> 
> This should be fixed in the latest push =] (the ppsspp buttons / retroarch ppsspp core loading menu)
> 
> 
> 
> Adding pipeline threading to geometry shader sounds good. There’s an implementation by the author around pipelining in general, but is minimally used at the moment due to stability / hardware GPU compatibility with older iOS devices. If MoltenVK provides good features around this we could definitely add to this when it becomes release code =] (Which.. Citra / emuThreeds teams seem to be very much on top of, so we should get an update from the teams)

The changes for PPSSPP seems to only affect the retroarch core and not the standalone one.

rf2222222 commented:
> > > There is a PR in moltenvk for geometry shaders and I wonder if it’s possible to use it for emuthreeds. The dev told me it should help performance but I wonder how that would be here. The emulator already only supports iOS 16+ so it makes sense. [KhronosGroup/MoltenVK#1815](https://github.com/KhronosGroup/MoltenVK/pull/1815)
> > 
> > 
> > > 
> > 
> > 
> > > also some now psp buttons is now odd on touch ui.
> > 
> > 
> > This should be fixed in the latest push =] (the ppsspp buttons / retroarch ppsspp core loading menu)
> > Adding pipeline threading to geometry shader sounds good. There’s an implementation by the author around pipelining in general, but is minimally used at the moment due to stability / hardware GPU compatibility with older iOS devices. If MoltenVK provides good features around this we could definitely add to this when it becomes release code =] (Which.. Citra / emuThreeds teams seem to be very much on top of, so we should get an update from the teams)
> 
> The changes for PPSSPP seems to only affect the retroarch core and not the standalone one.

This should now be fixed in the latest push, which adds an option for specifying confirmation button (cross / circle) with cross/circle button assignment, with cross/circle touch ui button assignment aligned with retroarch ppsspp setting. =]

SternXD commented:
What's the difference between Provenance XL Under-Development and Release?

rf2222222 commented:
> What's the difference between Provenance XL Under-Development and Release?

Provenance-XL has everything, Provenance has subset of Provenance-XL cores (which makes the binary slightly smaller), Provenance Under Development has experimental cores used for development (cores just starting to be looked at for integration / development). 

I usually use the Provenance-XL Release =]

SternXD commented:
> > What's the difference between Provenance XL Under-Development and Release?
> 
> Provenance-XL has everything, Provenance has subset of Provenance-XL cores (which makes the binary slightly smaller), Provenance Under Development has experimental cores used for development (cores just starting to be looked at for integration / development).
> 
> I usually use the Provenance-XL Release =]

Alright, thank you that helps me understand more :)

SternXD commented:
May I ask what your discord tag is?

rf2222222 commented:
> May I ask what your discord tag is?

I don’t usually use discord, but it’s: 

R22#8415

Just let me know that you came from GitHub PR, if I’m online =]

ba-sil-is-k commented:
Thank you for fixing the importer bugs where conflicts couldn't be resolved to the correct systems like psx. However, even though now I can import psx games normally, the library is refreshed whenever the app is opened, and this refresh reads all .cue files as separate games so multidisc games can be imported as one game but when the library ends up refreshing eventually, they show up as separate games again. So far I haven't found a reliable way to prevent this (although one game does stay as it should for some reason). I sent some screenshots and maybe more explanation to your discord tag, please let me know if you would rather I send them here.

rf2222222 commented:
> Thank you for fixing the importer bugs where conflicts couldn't be resolved to the correct systems like psx. However, even though now I can import psx games normally, the library is refreshed whenever the app is opened, and this refresh reads all .cue files as separate games so multidisc games can be imported as one game but when the library ends up refreshing eventually, they show up as separate games again. So far I haven't found a reliable way to prevent this (although one game does stay as it should for some reason). I sent some screenshots and maybe more explanation to your discord tag, please let me know if you would rather I send them here.

This should be fixed in the latest push =] Refreshing / Resetting the library from Settings will rebuild the game library with multi-disc support

ba-sil-is-k commented:
Thanks, it works great now! Quick question, I’m assuming these are disc numbers for the games, should they include the m3u (just asking because these games have one less discs than the number displayed?
![image](https://github.com/Provenance-Emu/Provenance/assets/134322971/6710408f-e94d-4087-9ac3-ee2d4dd8d32b)

rf2222222 commented:
> Thanks, it works great now! Quick question, I’m assuming these are disc numbers for the games, should they include the m3u (just asking because these games have one less discs than the number displayed? ![image](https://user-images.githubusercontent.com/134322971/266784569-6710408f-e94d-4087-9ac3-ee2d4dd8d32b.png)

This should be fixed in the latest push =] This adds m3u file filter from the count, along with new 'Choose Disc' option that appears when long pressing the game that allows selection of the cue / m3u file to launch.

ba-sil-is-k commented:
The disc count now works correctly and the new feature works too. Thx!

ba-sil-is-k commented:
Hey I know it may not be the core you’re most prioritizing, just wanted to let you know that the mame core does indeed work but I have noticed some glitches. Firstly the library treats the bios zip as a game and shows it and second, controls seem to be inverted in gameplay. 
![IMG_1280](https://github.com/Provenance-Emu/Provenance/assets/134322971/e929fd5e-bda4-4b24-89b2-461ad9dd4e54)
![IMG_1279](https://github.com/Provenance-Emu/Provenance/assets/134322971/7cdb955e-f345-4973-aea2-88b58ccf33bd)

ba-sil-is-k commented:
Sorry wrong picture
![image](https://github.com/Provenance-Emu/Provenance/assets/134322971/09149a70-f92a-4d38-a116-fc0146fb90bd)

denisoster commented:
`Reason: tried: '/usr/lib/swift/PVLogging.framework/PVLogging' (no such file, not in dyld cache), '/private/var/containers/Bundle/Application/B8CF7B45-42F1-43B2-9E6E-FE0A0EE795DD/ProvenanceTV.app/Frameworks/PVLogging.framework/PVLogging' (no such file), '/usr/lib/swift/PVLogging.framework/PVLogging' (no such file, not in dyld cache), '/private/var/containers/Bundle/Application/B8CF7B45-42F1-43B2-9E6E-FE0A0EE795DD/ProvenanceTV.app/Frameworks/PVLogging.framework/PVLogging' (no such file), '/System/Library/Frameworks/PVLogging.framework/PVLogging' (no such file, not in dyld cache)
(terminated at launch; ignore backtrace)`

rf2222222 commented:
> Hey I know it may not be the core you’re most prioritizing, just wanted to let you know that the mame core does indeed work but I have noticed some glitches. Firstly the library treats the bios zip as a game and shows it and second, controls seem to be inverted in gameplay. ![IMG_1280](https://user-images.githubusercontent.com/134322971/266888156-e929fd5e-bda4-4b24-89b2-461ad9dd4e54.jpeg) ![IMG_1279](https://user-images.githubusercontent.com/134322971/266888157-7cdb955e-f345-4973-aea2-88b58ccf33bd.jpeg)

This should be fixed in the latest push =] There is now a provenance controller on the NeoGeo / MAME cores.

rf2222222 commented:
> Hardware Model: ` AppleTV5,3` OS Version: `Apple TVOS 16.6 (20M73)`
> 
> ```
> Exception Type:  EXC_CRASH (SIGABRT)
> Exception Codes: 0x0000000000000000, 0x0000000000000000
> Termination Reason: DYLD 1 Library missing
> Library not loaded: @rpath/PVLogging.framework/PVLogging
> Referenced from: <CCC1E191-8957-3570-968D-7361FB53E9F3> /Volumes/VOLUME/*/ProvenanceTV.app/Provenance
> Reason: tried: '/usr/lib/swift/PVLogging.framework/PVLogging' (no such file, not in dyld cache), '/private/var/containers/Bundle/Application/B8CF7B45-42F1-43B2-9E6E-FE0A0EE795DD/ProvenanceTV.app/Frameworks/PVLogging.framework/PVLogging' (no such file), '/usr/lib/swift/PVLogging.framework/PVLogging' (no such file, not in dyld cache), '/private/var/containers/Bundle/Application/B8CF7B45-42F1-43B2-9E6E-FE0A0EE795DD/ProvenanceTV.app/Frameworks/PVLogging.framework/PVLogging' (no such file), '/System/Library/Frameworks/PVLogging.framework/PVLogging' (no such file, not in dyld cache)
> (terminated at launch; ignore backtrace)
> ```
> 
> my method resign [fastlane script](https://gist.github.com/denisoster/808c9965ce9cc4bca3233d491c9a3e61) because wiki methods not working for me
> 
> release version works well(sega genesis/ps1 with xbox controller tested )

This should be fixed in the latest push =]

denisoster commented:
@rf2222222 

> Added Volume Support to RetroArch, Emuthreeds, Play, etc

Does this mean that I can store and run games from external drive? My Apple TV has type-c port

rf2222222 commented:
> @rf2222222
> 
> > Added Volume Support to RetroArch, Emuthreeds, Play, etc
> 
> Does this mean that I can store and run games from external drive? My Apple TV has type-c port

The implementations for these are for iOS, so to play on TV, it'd have to be through AirPlay on an iOS device..

denisoster commented:
@rf2222222 

> Volume Support 

I don't understand what functionality is hidden behind the words

rf2222222 commented:
> @rf2222222
> 
> > Volume Support
> 
> I don't understand what functionality is hidden behind the words

This is the audio volume adjustment functionality on iOS devices, which is independent of the actual hardware volume, so if there are multiple audio streaming from different apps, the game volume could adjusted up/down during mixed play. (e.g. listening to di.fm while playing..etc =])

denisoster commented:
> > @rf2222222
> > > Volume Support
> > 
> > 
> > I don't understand what functionality is hidden behind the words
> 
> This is the audio volume adjustment functionality on iOS devices, which is independent of the actual hardware volume, so if there are multiple audio streaming from different apps, the game volume could be adjusted up/down during mixed play. (e.g. listening to di.fm while playing..etc =])

Thanks for the explanation =) 

Is it possible to implement storage and launch roms from external media(Apple TV has type-c port)?

rf2222222 commented:
> > > @rf2222222
> > > > Volume Support
> > > 
> > > 
> > > I don't understand what functionality is hidden behind the words
> > 
> > 
> > This is the audio volume adjustment functionality on iOS devices, which is independent of the actual hardware volume, so if there are multiple audio streaming from different apps, the game volume could be adjusted up/down during mixed play. (e.g. listening to di.fm while playing..etc =])
> 
> Thanks for the explanation =)
> 
> Is it possible to implement storage and launch roms from external media(Apple TV has type-c port)?

Yes, if you create a Pull Request, I think Joe will take a look ..etc =] Everyone's welcome to push code here

rf2222222 commented:
> Build worked!
> 
> Hardware Model: `AppleTV5,3` OS Version: `Apple TVOS 16.6 (20M73)`
> 
> GameCube, Dreamcast, PSP not supported? Bios installed, game correct detected but has message "No core found for game system GameCube, Dreamcast, PSP"

Great =] All the cores I implemented here (PSP, PS2, Wii, GameCube, RetroArch, 3DS, etc) specifically is targeted at iOS, since I personally don’t have an AppleTV device here and use Provenance to play games on my iPad / iPhone =] 
Perhaps some day if I get an AppleTV and no one else has implemented the support, I’ll do it at that time. So the best option right now to play these on TV is to use the iOS version with AirPlay on TV

gh0sti commented:
> > Build worked!
> > Hardware Model: `AppleTV5,3` OS Version: `Apple TVOS 16.6 (20M73)`
> > GameCube, Dreamcast, PSP not supported? Bios installed, game correct detected but has message "No core found for game system GameCube, Dreamcast, PSP"
> 
> Great =] All the cores I implemented here (PSP, PS2, Wii, GameCube, RetroArch, 3DS, etc) specifically is targeted at iOS, since I personally don’t have an AppleTV device here and use Provenance to play games on my iPad / iPhone =] Perhaps some day if I get an AppleTV and no one else has implemented the support, I’ll do it at that time. So the best option right now to play these on TV is to use the iOS version with AirPlay on TV

Since main dev Joe is absent and no longer working on this as far as we know and the Original dev Jasarien is letting this project go are you looking to fork over and just start maintaining this project yourself?

rf2222222 commented:
> > > Build worked!
> > > Hardware Model: `AppleTV5,3` OS Version: `Apple TVOS 16.6 (20M73)`
> > > GameCube, Dreamcast, PSP not supported? Bios installed, game correct detected but has message "No core found for game system GameCube, Dreamcast, PSP"
> > 
> > 
> > Great =] All the cores I implemented here (PSP, PS2, Wii, GameCube, RetroArch, 3DS, etc) specifically is targeted at iOS, since I personally don’t have an AppleTV device here and use Provenance to play games on my iPad / iPhone =] Perhaps some day if I get an AppleTV and no one else has implemented the support, I’ll do it at that time. So the best option right now to play these on TV is to use the iOS version with AirPlay on TV
> 
> Since main dev Joe is absent and no longer working on this as far as we know and the Original dev Jasarien is letting this project go are you looking to fork over and just start maintaining this project yourself?

I’ll just push the code I have here, to make changes I notice as I use the app. I’m contributing to maximize the enjoyment I get from the app, but I’m not looking to start my own version of it.

denisoster commented:
Please add dreamcast and gamecube for AppleTV

rf2222222 commented:
> Please add dreamcast and gamecube for AppleTV

I personally don’t own an AppleTV device, so the best bet is to build the sources for AppleTV =] :
Dreamcast (Flycast): https://github.com/flyinghead/flycast
GameCube / Wii (Dolphin iOS): https://github.com/OatmealDome/dolphin-ios

denisoster commented:
Please add support tv17

rf2222222 commented:
> Please add support tv17

The "ProvenanceTV - Release" schema should run on tvOS17. =]

rf2222222 commented:
@JoeMatt Now that iOS 17 is released to general public, this has all the fixes related to iOS16 -> iOS17 that fixes various crash issues related to this transition, so now’s the perfect timing to get this merged =]

gh0sti commented:
> @JoeMatt Now that iOS 17 is released to general public, this has all the fixes related to iOS16 -> iOS17 that fixes various crash issues related to this transition. I’m done with this round of changes related to iOS17 / 3DS Core / PSP updates ..etc so this is all good to go to get merged =]

do you have one last precompiled ipa until Joe can come around and do a merge to branch?

rf2222222 commented:
> > @JoeMatt Now that iOS 17 is released to general public, this has all the fixes related to iOS16 -> iOS17 that fixes various crash issues related to this transition. I’m done with this round of changes related to iOS17 / 3DS Core / PSP updates ..etc so this is all good to go to get merged =]
> 
> do you have one last precompiled ipa until Joe can come around and do a merge to branch?

The last build, https://github.com/Provenance-Emu/Provenance/pull/2210#issuecomment-1731973251 should be the latest =]

jwoo92 commented:
iOS/iPadOS 17.1 introduces support for the [Nintendo 64 controller](https://www.nintendo.com/us/store/products/nintendo-64-controller/).

It appears only the `start` button input registers in Provenance. Would someone be kind enough to take a look at working in support within this PR? Thanks.

Spidy123222 commented:
> > Please add support tv17
> 
> 
> 
> The "ProvenanceTV - Release" schema should run on tvOS17. =] 

Emuthreeds renamed to limon and has been updating more recently. Do you think this pr will get updated to reflect the changes?

rf2222222 commented:
> > > Please add support tv17
> > 
> > 
> > 
> > The "ProvenanceTV - Release" schema should run on tvOS17. =] 
> 
> Emuthreeds renamed to limon and has been updating more recently. Do you think this pr will get updated to reflect the changes? https://github.com/emuPlace/Limon

Seems tvOS is very popular request. I might get it in the future if I become popular enough with Apple for them to send me these devices to review as they come out with it =]

Actually talked to Joe about getting this merged so I could put in new features, but haven't heard back. If you get a hold of him and could get him to merge this PR, I could look at it as next Pr =]

Spidy123222 commented:
> > > > Please add support tv17
> > > 
> > > 
> > > 
> > > The "ProvenanceTV - Release" schema should run on tvOS17. =] 
> > 
> > Emuthreeds renamed to limon and has been updating more recently. Do you think this pr will get updated to reflect the changes? https://github.com/emuPlace/Limon
> 
> Seems tvOS is very popular request. I might get it in the future if I become popular enough with Apple for them to send me these devices to review as they come out with it =]
> 
> Actually talked to Joe about getting this merged so I could put in new features, but haven't heard back. If you get a hold of him and could get him to merge this PR, I could look at it as next Pr =]

Yeah us at sidestore don't have direct access to Joe as he's part of project also. I'll message him on discord but it could be a while. There was a project started by some people called genesis to maintain a experimental version for the meantime while provenance has been stuck in place for most part for a while. If joematt reads this, I suggest adding people to be able to maintain the app while being gone. TvOS I can see why it's popular. Though I could see it having problems with game size.

rf2222222 commented:
> > > > > Please add support tv17
> > > > 
> > > > 
> > > > The "ProvenanceTV - Release" schema should run on tvOS17. =]
> > > 
> > > 
> > > Emuthreeds renamed to limon and has been updating more recently. Do you think this pr will get updated to reflect the changes? https://github.com/emuPlace/Limon
> > 
> > 
> > Seems tvOS is very popular request. I might get it in the future if I become popular enough with Apple for them to send me these devices to review as they come out with it =]
> > Actually talked to Joe about getting this merged so I could put in new features, but haven't heard back. If you get a hold of him and could get him to merge this PR, I could look at it as next Pr =]
> 
> Yeah us at sidestore don't have direct access (outside of GitHub/discord) to Joe as he's part of project also. I'll message him on discord but it could be a while. There was a project started by some people called genesis to maintain a experimental version for the meantime while provenance has been stuck in place for most part for a while. If joematt reads this, I suggest adding people to be able to maintain the app while being gone. TvOS I can see why it's popular. Though I could see it having problems with game size.

Yeah can you let them know I’ll maintain Provenance / get these PRs merged / get to bug fixing if they’re willing to grant the admin permission. (The donation / proceeds generated by Provenance are still theirs to keep) It’s been a while =]

JoeMatt commented:
This builds for me on XCode Version 15.1 (15C65)

Amazing work. I want to check if this merges well with @dnicolson 's PRs too and see if merge/rebase order matters.

Will try to keep things as clean as possible. 

Since the amount of changes is massive, but it works, I'm leaning towards restarting all my previous side work on SPM and stuff later based on this and dni's changes too and then also reviewing if i want to 'correct' any changes after the fact on 'develop' rather than take days to review the pr's one by one and merging one by one etc

pokemon-card commented:
Any way to increase fps on emuthreeds?

rf2222222 commented:
> Any way to increase fps on emuthreeds?

@pokemon-card The project repository has latest and code base that is always getting updated. This is at:

https://emuthreeds.com/  
https://github.com/emuplace/limon
